ts/nix.ts

```diff
@@ -195,13 +195,16 @@ export function nixStrLit(
   };
 }
 
-export function unlinesNixStrings(list: Array<NixExpression>): NixExpression {
+export function joinNixStrings(
+  separator: string,
+  list: Array<NixExpression>,
+): NixExpression {
   return {
     [__nixExpressionTag]: null,
     type: "strLit",
     str: {
       initial: "",
-      rest: list.map((expr) => [expr, "\n"]),
+      rest: list.map((expr, i) => [expr, i < list.length - 1 ? separator : ""]),
```

Comment:
I ended up doing something very similar, except for `NixExpression`s vs `NixStrLit`s here: https://github.com/garnix-io/garn/pull/430/files#diff-fb4f9f2ab46707d603912af39f9d791c92eb585508b43f11b86e588c7c88513eR81-R86

I wonder if it makes sense to attempt to consolidate these somehow (when one of our PRs makes it in) or if the fact one operates on raw nix expressions and the other operates on string literals means they stay separate.

Reply:
Done in a9ba3b57.